sycl/include/CL/sycl/detail/backend_traits_hip.hpp

```diff
@@ -0,0 +1,136 @@
+//===------- backend_traits_hip.hpp - Backend traits for HIP ---*-C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines the specializations of the sycl::detail::interop,
+// sycl::detail::BackendInput and sycl::detail::BackendReturn class templates
+// for the HIP backend but there is no sycl::detail::InteropFeatureSupportMap
+// specialization for the HIP backend.
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <CL/sycl/accessor.hpp>
+#include <CL/sycl/context.hpp>
+#include <CL/sycl/detail/backend_traits.hpp>
+#include <CL/sycl/device.hpp>
+#include <CL/sycl/event.hpp>
+#include <CL/sycl/kernel_bundle.hpp>
+#include <CL/sycl/queue.hpp>
+
+typedef int HIPdevice;
+typedef struct ihipCtx_t *HIPcontext;
```

Comment:
I can see that [HIP API for managing context](https://rocmdocs.amd.com/en/latest/Programming_Guides/HIP_API_Guide.html#deprecated-hip-apis) is deprecated. Does it make sense to introduce interop API for deprecated backend type?

Reply:
The current [PI_HIP](https://github.com/intel/llvm/blob/sycl/sycl/plugins/hip/pi_hip.hpp#L133) backend implements the HIP native context handle using hipCtx_t.

Hence the template specification in the backend_traits_hip.hpp also tries to align with the same (ref: [hip_runtime.h](https://github.com/ROCm-Developer-Tools/HIP/blob/develop/include/hip/hip_runtime_api.h#L536)).

I do agree that it'd be beneficial to eliminate the need for the deprecated APIs. However, there doesn't seem to be any other APIs for dealing with context objects. Do provide any suggestions in this regard, if I misread anything.

Reply:
I see, thanks for explaining this.